src/services/document-service.spec.js

```diff
@@ -1038,4 +1048,114 @@ describe('document-service', () => {
       });
     });
   });
+
+  describe('createRegenerateDocumentsBatch', () => {
+    let createdDocument;
+    let result;
+    let revision;
+    beforeEach(async () => {
+      revision = {
+        title: 'Title',
+        slug: 'my-doc',
+        language: 'en',
+        sections: [],
+        tags: ['tag-1']
+      };
+      result = await sut.createNewDocumentRevision({ doc: revision, user });
+      createdDocument = await db.documents.findOne({ key: result.key });
+    });
+
+    describe('when a running batch already exists', () => {
+      beforeEach(async () => {
+        await sut.createRegenerateDocumentsBatch(user);
+      });
+
+      it('should return null', async () => {
+        const batch = await sut.createRegenerateDocumentsBatch(user);
+        expect(batch).toBeNull();
+      });
+    });
+
+    describe('when a new batch is created', () => {
+      let batch = null;
+      beforeEach(async () => {
+        batch = await sut.createRegenerateDocumentsBatch(user);
+      });
+
+      it('should create the new batch', () => {
+        expect(batch).toEqual({
+          _id: expect.stringMatching(/\w+/),
+          createdBy: user._id,
+          createdOn: new Date(),
+          completedOn: null,
+          batchType: BATCH_TYPE.regenerateDocuments,
+          batchParams: {},
+          errors: []
+        });
+      });
+
+      it('should create the according tasks', async () => {
+        const tasks = await taskStore.find({ batchId: batch._id });
+        expect(tasks).toEqual([
+          {
+            _id: expect.stringMatching(/\w+/),
+            batchId: batch._id,
+            taskType: TASK_TYPE.regenerateDocument,
+            processed: false,
+            attempts: [],
+            taskParams: {
+              key: createdDocument.key
+            }
+          }
+        ]);
+      });
+    });
+  });
+
+  describe('regenerateDocument', () => {
+    const lock = { _id: 'mylock' };
+
+    let regeneratedDocument;
+    let result;
+    let revision;
+
+    beforeEach(async () => {
+      sandbox.stub(documentLockStore, 'takeLock').resolves(lock);
+      sandbox.stub(documentLockStore, 'releaseLock');
+      revision = {
+        title: 'Title',
+        slug: 'my-doc',
+        language: 'en',
+        sections: [],
+        tags: ['tag-1']
+      };
+
+      result = await sut.createNewDocumentRevision({ doc: revision, user });
```

Comment:
This `result` confused me a bit, I thought it's the result we need to assert, as per our test pattern.
Could we please call this `document`?

Reply:
Renamed, it was confusing, sorry